server/datastore/mysql/software.go

```diff
@@ -514,12 +505,17 @@ func selectSoftwareSQL(opts fleet.SoftwareListOptions) (string, []interface{}, e
 
 	ds = ds.GroupBy(
 		"s.id",
-		"scv.cpe_id",
+		"s.name",
```

Comment:
Why are you adding all the columns to the group by?

Reply:
https://github.com/fleetdm/fleet/pull/6587#discussion_r921542296

Excluding columns in the `GROUP BY` does not comply with SQL-92 also (at least for me) it makes the query easier to read because you don't need to keep track of functional dependencies in case you want to modify the query.

Reply:
From https://dev.mysql.com/doc/refman/5.7/en/group-by-handling.html

> SQL:1999 and later permits such nonaggregates per optional feature T301 if they are functionally dependent on GROUP BY columns: If such a relationship exists between name and custid, the query is legal. This would be the case, for example, were custid a primary key of customers.
> 
> MySQL 5.7.5 and later implements detection of functional dependence. If the [ONLY_FULL_GROUP_BY](https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sqlmode_only_full_group_by) SQL mode is enabled (which it is by default), MySQL rejects queries for which the select list, HAVING condition, or ORDER BY list refer to nonaggregated columns that are neither named in the GROUP BY clause nor are functionally dependent on them. (Before 5.7.5, MySQL does not detect functional dependency and [ONLY_FULL_GROUP_BY](https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sqlmode_only_full_group_by) is not enabled by default. For a description of pre-5.7.5 behavior, see the [MySQL 5.6 Reference Manual](https://dev.mysql.com/doc/refman/5.6/en/sql-mode.html).)

While it doesn't comply with SQL-92, it does comply with SQL:1999. I don't think we are following a particular standard, but it is worth noting.

Is there a performance impact to adding the columns to the `GROUP BY`?

Reply:
> Is there a performance impact to adding the columns to the GROUP BY?

There shouldn't any performance impact - as a side note this query was changed in https://github.com/fleetdm/fleet/pull/6879 to address its performance issues.